pkg/agent/catalog/catalog.go

```diff
@@ -135,7 +131,28 @@ func Load(ctx context.Context, config Config) (*Repository, error) {
 	p.KeyManager.KeyManager = keymanager_telemetry.WithMetrics(p.KeyManager.KeyManager, config.Metrics)
 
 	return &Repository{
-		Catalog: p,
-		Closer:  closer,
+		Catalog: &Plugins{
+			KeyManager:        p.KeyManager,
+			NodeAttestor:      nodeattestor.V0(p.NodeAttestor, p.NodeAttestor),
+			WorkloadAttestors: p.WorkloadAttestors,
+		},
+		Closer: closer,
 	}, nil
 }
+
+// versionedPlugins is a temporary struct to hold the raw plugin interfaces so
```

Comment:
Want to emphasize that this is temporary. It won't be necessary i the new catalog system.